tree/dataframe/src/RNTupleDS.cxx

```diff
@@ -30,6 +32,43 @@
 
 namespace ROOT {
 namespace Experimental {
+namespace Detail {
+struct RNTupleColumnReader final : public ROOT::Detail::RDF::RColumnReaderBase {
+   using RFieldBase = ROOT::Experimental::Detail::RFieldBase;
+   using RFieldValue = ROOT::Experimental::Detail::RFieldValue;
+   using RPageSource = ROOT::Experimental::Detail::RPageSource;
+
+   std::unique_ptr<RFieldBase> fField;
+   RFieldValue fValue;
+   Long64_t fLastEntry; ///< Last entry number that was read
+
+   std::unique_ptr<RFieldBase> MakeField(const std::string &colName, RPageSource &source)
+   {
+      const auto &descriptor = source.GetDescriptor();
+      const auto fieldId = descriptor.FindFieldId(colName);
+      const auto &fieldDescriptor = descriptor.GetFieldDescriptor(fieldId);
+      const auto typeName = fieldDescriptor.GetTypeName();
+      auto fieldBasePtr = Detail::RFieldBase::Create(fieldDescriptor.GetFieldName(), typeName);
+      Detail::RFieldFuse::ConnectRecursively(fieldId, source, *fieldBasePtr);
+      return std::unique_ptr<RFieldBase>(fieldBasePtr);
+   }
+
+public:
```

Comment:
This should only be necessary if `RNTupleColumnReader` is a `class`

Reply:
indeed, I forgot to change it to a class! :+1: fixed